src/store/CategoricalDisplayStore.js

```diff
@@ -215,15 +233,15 @@ export default class CategoricalDisplayStore {
     }
     setFavorite(item) {
         if (this.favoritesMap[item.itemId]) {
-            this.favoritesCategory.data = this.favoritesCategory.data.filter(x => x.itemId !== item.itemId);
+            this.favoritesCategory.data = this.favoritesCategory.data.filter(x => x && x.itemId !== item.itemId && x !== item.itemId);
```

Comment:
nitpick:  please don't use single letter variables like `x` :-)
If you need to shorten the name use 3letter variables like:
`rows.forEach(row => { });` or `favoritesCategory.data.forEach(cat => { })` or `favorites.forEach(fav => { })`.

And if you really need single letter variables which are handy in lambdas use the first letter like `rows.forEach(r => { })` or `favorites.forEach(f => {})`

Reply:
sure.
should I modify this part?

Reply:
@shayanypn yes please :-)